# 1주차 예상문제 1

데이터 출처 : https://data.kma.go.kr/cmmn/static/staticPage.do?page=intro

데이터 설명 : 2020년도 이화동(서울) , 수영동(부산)의 시간단위의 기온과 강수량

데이터 url : https://raw.githubusercontent.com/Datamanim/datarepo/main/weather/weather2.csv  

출처 :  1주차 예상문제 1 (이기적 스터디 카페 / 컴활,정보처리,빅분기)

In [1]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/weather/weather2.csv")

In [2]:
df

time  이화동강수  이화동기온  수영동강수  수영동기온
0     2020-01-01 00:00:00    0.1   -3.9       0.0       2.5
1     2020-01-01 01:00:00    0.0   -3.1       0.0       3.4
2     2020-01-01 02:00:00    0.0   -1.8       0.0       3.8
3     2020-01-01 03:00:00    0.0   -0.8       0.0       4.6
4     2020-01-01 04:00:00    0.0   -0.1       0.0       5.1
...                   ...    ...    ...       ...       ...
8779  2020-12-31 19:00:00    0.0   -9.5       0.0      -4.5
8780  2020-12-31 20:00:00    0.0   -9.5       0.0      -5.0
8781  2020-12-31 21:00:00    0.0   -9.3       0.0      -6.5
8782  2020-12-31 22:00:00    0.0   -9.2       0.0      -4.6
8783  2020-12-31 23:00:00    0.0   -7.8       0.0      -0.2

[8784 rows x 5 columns]

## Q1. 여름철(6월, 7월, 8월) 이화동이 수영동보다 높은 기온을 가진 시간대는 몇개인가?

In [3]:
df['time'] = pd.to_datetime(df['time'])
summer = df.loc[df['time'].dt.month.isin([6,7,8])].reset_index(drop=True)
summer

time  이화동강수      이화동기온  수영동강수   수영동기온
0    2020-06-01 00:00:00    0.0  21.100000       0.0  24.799999
1    2020-06-01 01:00:00    0.0  22.400000       0.0  23.500000
2    2020-06-01 02:00:00    0.0  23.100000       0.0  23.100000
3    2020-06-01 03:00:00    0.0  23.100000       0.0  25.400000
4    2020-06-01 04:00:00    0.0  23.799999       0.0  24.500000
...                  ...    ...        ...       ...        ...
2203 2020-08-31 19:00:00    0.0  24.299999       0.0  24.100000
2204 2020-08-31 20:00:00    0.0  23.900000       0.0  24.200001
2205 2020-08-31 21:00:00    0.0  24.000000       0.0  24.600000
2206 2020-08-31 22:00:00    0.0  24.600000       0.0  24.900000
2207 2020-08-31 23:00:00    0.0  25.000000       0.0  26.900000

[2208 rows x 5 columns]

In [4]:
answer = len(summer.loc[summer['이화동기온'] > summer['수영동기온']])
answer

1415

## Q2. 이화동과 수영동의 최대강수량의 시간대를 각각 구하여라

- 나의 풀이

In [5]:
print(df[df['이화동강수'] == df['이화동강수'].max()]['time'].values[0])

2020-09-30T09:00:00.000000000


In [6]:
print(df[df['수영동강수'] == df['수영동강수'].max()]['time'].values[0])

2020-07-23T12:00:00.000000000


# 1주차 예상문제 2

데이터 출처 : https://www.kaggle.com/shubh0799/churn-modelling 에서 변형

데이터 설명 :  고객의 신상정보 데이터를 통한 회사 서비스 이탈 예측 (종속변수 : Exited)

Data url : https://raw.githubusercontent.com/Datamanim/datarepo/main/churn/train.csv

출처 : 1주차 예상문제 2 (이기적 스터디 카페 / 컴활,정보처리,빅분기)

## Q1.  남성 이탈(Exited)이 가장 많은 국가(Geography)는 어디이고 이탈 인원은 몇명인가?

In [7]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/Datamanim/datarepo/main/churn/train.csv")

In [8]:
df

RowNumber  CustomerId     Surname  CreditScore Geography  Gender  Age  \
0          6842    15793491  Cherkasova          714   Germany    Male   26   
1          8963    15607874       Keane          687    France    Male   38   
2          7047    15737627      Rivero          589   Germany  Female   20   
3          7503    15697844  Whitehouse          721     Spain  Female   32   
4          3439    15722404   Carpenter          445    France  Female   30   
...         ...         ...         ...          ...       ...     ...  ...   
7994       1607    15714080      Goliwe          566   Germany  Female   40   
7995       7744    15741643      Chiang          777   Germany    Male   35   
7996       4231    15566660        Cole          670    France  Female   41   
7997       8870    15733597        Y?an          669    France  Female   41   
7998        523    15595766       Watts          527     Spain    Male   37   

      Tenure    Balance  NumOfProducts  HasCrCard  IsActiveMember  \
0          3  119545.48              2          1               0   
1          0  144450.58              1          0               1   
2          2  121093.29              2          1               0   
3         10       0.00              1          1               0   
4          3       0.00              2          1               1   
...      ...        ...            ...        ...             ...   
7994       2   97001.36              2          1               0   
7995       7  122917.69              1          1               1   
7996      10       0.00              3          1               0   
7997       0  150219.41              2          0               0   
7998       5   93722.73              2          1               1   

      EstimatedSalary  Exited  
0            65482.94       0  
1           137276.83       0  
2             3529.72       0  
3           136119.96       1  
4           127939.19       0  
...               ...     ...  
7994        154486.01       0  
7995         76169.68       0  
7996         81602.02       0  
7997        107839.03       0  
7998        139093.73       0  

[7999 rows x 14 columns]

In [9]:
df[df['Gender'] == 'Male'].groupby(['Geography']).sum()['Exited']

Geography
France     285
Germany    287
Spain      137
Name: Exited, dtype: int64

=> Germany, 287명

## Q2. 카드를 소유(HasCrCard ==1)하고 있으면서 활성멤버(IsActiveMember ==1) 인 고객들의 평균나이는? 

In [10]:
df[(df['HasCrCard'] == 1) & (df['IsActiveMember'] == 1)]['Age'].mean()

39.61019283746556

## Balance 값이 중간값 이상을 가지는 고객들의 CreditScore의 표준편차를 구하여라

In [11]:
df[df['Balance'] >= df['Balance'].median()]['CreditScore'].std()

97.29451567120783